File: recursion/cpp/BinarySearch.cpp
#include<bits/stdc++.h>

using namespace std;

bool binary_search(int a[],int target,int low,int high)
{
	if(low>high)
	return false;
	int mid=low+(high-low)/2;
	if(a[mid]==target)
	return true;
	if(a[mid]>target)
	return binary_search(a,target,low,mid-1);
	else return binary_search(a,target,mid+1,high);
}

int main()
{
	int a[5]={2, 3, 4, 10, 40};				// the array must be sorted to apply binary search
	int n=5;
	int target=15;								//target element to be found
	bool present=binary_search(a,target,0,n-1);
	if(present)									// if element found print yes else no
	{
		cout<<"Yes, the element is present in the array\n";
	}
	else
	{
		cout<<"No, the element is not present in the array\n";
	}
	return 0;
}

File: binary_search.java
import java.util.*;
class binary_search
{

int search(int a[], int ub, int lb, int m)
{
int mid;

if(lb <= ub)
{
	mid = (ub+lb)/2;

	if(m > a[mid])
		return search(a, ub, mid+1, m);

	else if(m < a[mid])
		return search(a, mid-1, lb, m);

	else 
	{
		System.out.println("Element found at index "+mid);
		return mid;
	}
}

return -1;
}	
	
public static void main(String args[])
{
Scanner sc=new Scanner(System.in);
binary_search ob=new binary_search();

System.out.println("Enter the number of elements");
int n=sc.nextInt();
int a[]=new int[n];

System.out.println("Enter a sorted array");
for(int i=0; i<n; i++)
	a[i] = sc.nextInt();

System.out.println("Enter the search element");
int m=sc.nextInt();

int res = ob.search(a, n, 0, m);

if(res == -1)
	System.out.println("Element not found");
}
}